### ISSUE TITLE
terraform destroy errors with config map

### ISSUE THREAD

**Issue by marcstreeter** [hashicorp/learn-terraform-provision-eks-cluster#31]
Version:
terraform 0.14.5

Platform:
Ubuntu 20.04 (within virtual box on a windows machine)

After deploying (successfully) I attempted to remove with `terraform destroy`.  It failed once with:

```
Error: Unauthorized
```

I retried destroying (without changing anything) and it proceeded to delete all except for `module.eks.kubernetes_config_map.aws_auth[0]`, even after multiple attempts it persisted giving me the same error:

```
Error: Delete "http://localhost/api/v1/namespaces/kube-system/configmaps/aws-auth": dial tcp 127.0.0.1:80: connect: connection refused
```

I'm not sure but what I could find was another issue that has a similar error but on deploy: https://github.com/hashicorp/learn-terraform-provision-eks-cluster/issues/12

**Comment by marcstreeter** [comment]
error comes only after working with the cluster, most likely like loadbalancers they must be cleaned out in order for terraform to destroy without errors

**Comment by marcstreeter** [comment]
k so it happens even if I don't use it (just so happen to bring up a cluster and trash it again). Any ideas?

**Comment by marcstreeter** [comment]
Here is the full output if it helps anyone:

```
base ❯ terraform destroy                                                                                              

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # module.eks.kubernetes_config_map.aws_auth[0] will be destroyed
  - resource "kubernetes_config_map" "aws_auth" {
      - data = {
          - "mapAccounts" = jsonencode([])
          - "mapRoles"    = <<-EOT
                - "groups":
                  - "system:bootstrappers"
                  - "system:nodes"
                  "rolearn": "REMOVED-TEXT-HERE"
                  "username": "system:node:{{EC2PrivateDNSName}}"
            EOT
          - "mapUsers"    = jsonencode([])
        } -> null
      - id   = "kube-system/aws-auth" -> null

      - metadata {
          - generation       = 0 -> null
          - labels           = {
              - "app.kubernetes.io/managed-by" = "Terraform"
              - "terraform.io/module"          = "terraform-aws-modules.eks.aws"
            } -> null
          - name             = "aws-auth" -> null
          - namespace        = "kube-system" -> null
          - resource_version = "1361" -> null
          - uid              = "REMOVED-TEXT-HERE" -> null
        }
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Changes to Outputs:

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.
```
which after proceeding gives
```
module.eks.kubernetes_config_map.aws_auth[0]: Destroying... [id=kube-system/aws-auth]

Error: Delete "http://localhost/api/v1/namespaces/kube-system/configmaps/aws-auth": dial tcp 127.0.0.1:80: connect: connection refused```

**Comment by vsheffer** [comment]
I've had several different issues with destroying the cluster, this is the latest.  I'm trying to move a test cluster from one region to another and this is failing.  Not the highest priority, for me at least, but one I'd like to try to help resolve.

I know some of my cluster was destroyed, but now I can't even do a plan, so I'll have to check on the current state to try to discern the resources that remain.

Having used GKE I'm not a fan of the AWS EKS permissions requiring some interplay between IAM and K8s RBAC.  Based on the error that is what looks like the issue.

**Comment by admiral-hackbar** [comment]
I ran into the unauthorized issue as well recently. Short story is terraform v14 doesn't automatically refresh anymore before a destroy. Run `terraform refresh` before you destroy, and that should help.

These are the threads I found the answer on:
https://discuss.hashicorp.com/t/eks-cluster-not-destroyed-completely-getting-error-unauthorized/18569/7
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1162
